Assets/BossRoom/Scripts/Server/Game/State/ServerBossRoomState.cs

```diff
@@ -180,8 +182,18 @@ private void SpawnPlayer(ulong clientId, bool lateJoin)
             // if joining late, assign a random character to the persistent player
             if (lateJoin)
             {
-                persistentPlayer.NetworkAvatarGuidState.AvatarGuid.Value =
-                    m_AvatarRegistry.GetRandomAvatar().Guid.ToNetworkGuid();
+                SessionPlayerData? sessionPlayerData = SessionManager.Instance.GetPlayerData(clientId);
```

Comment:
Actually, on the SessionManager's side, it uses the clientID to get the guid, for which the mapping is updated each time a new client connects. But we could skip that step and simply get the guid here instead via ClientPrefs and pass that to the SessionManager

Reply:
Nevermind, it doesn't work that way since this code is executed on the host, so we don't have access to the guid through ClientPrefs, and will have to get it via the client ID anyways. Anyways this is fine since the mapping of client IDs and GUIDs is handled by the SessionManager. I just tested it with 2 clients, one via parrelsync and one via a build (I had to revert commit 7315fae to be able to test this) and no matter the order of connection/disconnection/reconnection they always reconnect as the right character.
However I also tested it with 2 clients via parrelsync and there was an issue, the modified guid (adding _Secondary) is not handled correctly and if both clients disconnect and reconnect, they currently both reconnect as the first client's character. I'll look into this.